### DISCUSSION TITLE
Is it possible to allow a combined word where the single word is not allowed?

### DISCUSSION THREAD

**Discussion by jugmac00** [get-woke/woke#159] [Q&A]
Hi,

thanks for this amazing tool!

In one project we do not allow `dummy`, but there is a reference to a class in an external library named  `_DummyTransport`.

Is it possible to configure `_DummyTransport` as an exception in the configuration file rather than doing an inline/next line exemption?

Thank you!

**Background**

Either it looks like this, where I additionally need to ignore exceeding line length (E501):
```
# For SFTP connections, 'transport' is actually a _DummyTransport  # wokeignore:rule=dummy  # noqa: E501
# instance. Neither _DummyTransport nor the protocol it wraps  # wokeignore:rule=dummy  # noqa: E501
# (filetransfer.FileTransferServer) support pausing.
```

or I need to add the next line ignore every other line which makes the comment harder to read:
```
# wokeignore:rule=dummy
# For SFTP connections, 'transport' is actually a _DummyTransport
# wokeignore:rule=dummy
# instance. Neither _DummyTransport nor the protocol it wraps
# (filetransfer.FileTransferServer) support pausing.
```

**Comment by cognitivegears** [comment]
Yes! You can do this by setting the word_boundary_end parameter to true for a rule. See the following link:

https://docs.getwoke.tech/rules/

With this set to true, a word like _DummyTransport would not be matched, while Dummy by itself would be. Also TheDummy would be matched as well - to only get Dummy by itself use word_boundary to specify both the start and end boundaries.

However - if this doesn't work for you, if you want *only* _DummyTransport to be excluded but other compound words be included then unfortunately that doesn't exist yet. I have a pull request for regular expression functionality that would do this, but I need to do a bit more work before that is supported. Hopefully soon though 🙂

**Reply by slyon** [reply]
@cognitivegears How about excluding `dummy` but allowing for `dummy-devices`? The word_boundary parameter doesn't seem to do the trick here, even though the hyphen `-` is an ASCII character...
